### DISCUSSION TITLE
ImageGlass 9.0.8.1208

### DISCUSSION THREAD

**Discussion by d2phap** [d2phap/ImageGlass#1751] [General]
<img width="640" alt="ImageGlass 9.0.8.1208" src="https://raw.githubusercontent.com/ImageGlass/releases/main/screenshots/v9.0/9.0_b1.webp">

[![Downloads](https://img.shields.io/github/downloads/d2phap/imageglass/9.0.8.1208/total?color=%232A7C91&label=downloads&style=for-the-badge)](https://imageglass.org/release/imageglass-9-0-8-1208-46)

Please support this project by making a donation or purchasing the Microsoft Store version:

<a href="https://www.microsoft.com/store/productId/9N33VZK3C7TH?cid=GitHubRelease&referrer=appbadge&source=GithubD2phap">
  <img height="69" src="https://user-images.githubusercontent.com/3154213/197671083-4099d80f-07fc-4af9-b80d-6357f0968347.svg" />
</a>

## 🪁 System Requirements
- Windows 10/11 64-bit, version 2004 or later
- [.NET Desktop Runtime 8 (64-bit)](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)
- [WebView2 Runtime 64-bit](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section)

## 📝 Changelog (since [ImageGlass 8.10](https://github.com/d2phap/ImageGlass/releases/tag/8.10.9.27))
Please read the blogpost: https://imageglass.org/news/introducing-the-new-imageglass-version-9-88

### 🔮 Improvements:
- MSI installer:
  + Added support for Windows Server 2019 x64 (#1711)
  + Added support for installing as system wide app for all users (#1708)
- Added support for auto-detecting WebView2 Runtime and showing a message if user does not have it installed (#1722, #1724)
- Changed the WebView2 user data location from the app config folder to `%LocalAppData%\ImageGlass\<version>\Webview2_Data\`
- Added WebView2 Runtime information in About dialog
- Added support for copying version information in About dialog (#1730)
- Improved image interpolation setting (#1716):
  + Always uses `NearestNeighbor` when zoom factor is 100%
  + Updated the setting `When zoom ≤ 100%` to `When zoom < 100%`
- Removed in-app message for all window modes (#1739)
- Added <kbd>+</kbd> and <kbd>-</kbd> numpad key together with <kbd>=</kbd>, <kbd>-</kbd> as default zooming hotkeys
- Added a message in app settings where user select `Current monitor profile` option for Color management to tell them ImageGlass does not auto-update monitor color profile when moving between monitors (#1709)
- Removed the Save/Save As context button on the toolbar when users edit the image
- Updated the default toolbar buttons to make it similar to version 8 (#1648)
- Renamed `Page navigation` tool to `Frame navigation` tool, and its menu name from `MnuPageNav` to `MnuFrameNav`
- Added `.avif` to the `SingleFrameFormats` (#1735)
- Enabled "Maximize" button in Window Fit mode

### 🐛 Bugfixes:
- Fixed an issue where the functionality of Zoom In and Zoom Out button on toolbar was swapped (#1704)
- Fixed an issue where the style of Full Screen button on toolbar was not updated after exiting Full Screen mode (#1729)
- Fixed an issue where user could not increase the slideshow interval (#1728)
- Fixed an issue where user could not run ImageGlass as Administrator because WebView2 could not write data to the installed directory (#1725)
- Fixed an issue where scroll wheel didn't work if "Scroll inactive windows when I hover over them" is disabled in Windows settings (#1723)
- Fixed an issue where the image was blurry at 100% zoom (#1701, #1716, #1736, #1737)
- Fixed an issue where ImageGlass window was flickering when manually resizing Window Fit mode (#1720)
- Fixed an issue where user could not minimize ImageGlass window in Window Fit mode (#1741)
- Fixed an issue where ARM64 build did not work (#1717)
- Fixed an issue where user could not zoom the SVG image with WebView2 by mouse wheel due to JSON parsing error (#1743)
- Fixed an issue where frame size was not render correctly when viewing separately
- Fixed an issue where users could not save tool settings
- Fixed an issue where image was blinking when switching between images
- Fixed an issue where setting `EnableImageTransition` did not apply for Slideshow mode
- Fixed an issue where image file created by Save As function did not appear in the Gallery

## 🟣 Other updates for ImageGlass 9.0
- Published [ImageGlass.Tools library v1.9070](https://github.com/ImageGlass/ImageGlass.Tools)
- Published [ExifGlass app v1.3](https://github.com/d2phap/ExifGlass)

## 🟣 New contributor:
@turbedi with PR #1746

**Full Changelog**: https://github.com/d2phap/ImageGlass/compare/9.0.7.1125...9.0.8.1208

<hr /><em>This discussion was created from the release <a href='https://github.com/d2phap/ImageGlass/releases/tag/9.0.8.1208'>ImageGlass 9.0.8.1208</a>.</em>

**Comment by egeres** [comment]
Amazing work!!! Thanks!! ❤❤❤

**Comment by pbjberg** [comment]
Hello.  We use ImageGlass in our Citrix VDI environment.  Since this update, our ImageGlass has stopped working.  We get the following error upon opening:

![2024-01-04_13-04-03](https://github.com/d2phap/ImageGlass/assets/155683327/0356ebcc-530a-4683-96f6-ee17175e0d58)

Pressing Continue causes the application to close.  We are also seeing the following errors in Event Viewer:

Application: ImageGlass.exe
CoreCLR Version: 8.0.23.53103
.NET Version: 8.0.0
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ComponentModel.Win32Exception (0xC000000D): Unknown error (0xc000000d)
   at DirectN.VerticalBlankTicker.Wait(Object state)

and

Faulting application name: ImageGlass.exe, version: 9.0.9.1208, time stamp: 0x65410000
Faulting module name: KERNELBASE.dll, version: 10.0.17763.4974, time stamp: 0x909c5ed9
Exception code: 0xe0434352
Fault offset: 0x00000000000349b9
Faulting process id: 0x249c
Faulting application start time: 0x01da3f5263578073
Faulting application path: C:\Program Files\ImageGlass\ImageGlass.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: 6b92d4ba-b681-4e08-b2fd-8973572a0ee1
Faulting package full name: 
Faulting package-relative application ID:

**Reply by pbjberg** [reply]
We also tried installing the .1230 update but the errors persisted.

**Reply by d2phap** [reply]
It's a known issue #1771, some are running fine though (#1711)

**Reply by pbjberg** [reply]
Thanks for the update.  It is odd, I am able to open it fine on the master image, but after I clone it, I get the error.  Also, I tried running it as admin from the clone.  Got the following errors:

![error1](https://github.com/d2phap/ImageGlass/assets/155683327/0a7dcf60-5452-4ee7-942e-dfc077457786)

![error2](https://github.com/d2phap/ImageGlass/assets/155683327/97ee3d98-8c3b-477a-b6c4-e49f86a87073)

**Reply by d2phap** [reply]
The error message says IG does not have write permission in the folder ...\AppData\Local\ImageGlass\... I don't know what you mean by "clone".

Anyway your IG version is outdated.
